posthog/hogql_queries/experiments/experiment_query_runner.py

```diff
@@ -0,0 +1,449 @@
+                                    right=ast.Field(chain=["events_after_exposure", "variant"]),
+                                    op=ast.CompareOperationOp.Eq,
+                                ),
+                            ]
+                        ),
+                        constraint_type="ON",
+                    ),
+                ),
+            ),
+            group_by=[
+                ast.Field(chain=["exposure_data", "variant"]),
+                ast.Field(chain=["exposure_data", "distinct_id"]),
+            ],
+        )
+
+        # Final results: One row per variant
+        # Columns: variant, num_users, total_sum, total_sum_of_squares
+        # Aggregates user metrics into final statistics used for significance calculations
+        experiment_variant_results_query = ast.SelectQuery(
+            select=[
+                ast.Field(chain=["metrics_per_user", "variant"]),
+                parse_expr("count(metrics_per_user.distinct_id) as num_users"),
+                parse_expr("sum(metrics_per_user.value) as total_sum"),
+                parse_expr("sum(power(metrics_per_user.value, 2)) as total_sum_of_squares"),
+            ],
+            select_from=ast.JoinExpr(table=metrics_aggregated_per_user_query, alias="metrics_per_user"),
+            group_by=[ast.Field(chain=["metrics_per_user", "variant"])],
+        )
+
+        return experiment_variant_results_query
+
+    def _evaluate_experiment_query(
+        self,
+    ) -> list[ExperimentVariantTrendsBaseStats] | list[ExperimentVariantFunnelsBaseStats]:
+        response = execute_hogql_query(
+            query=self._get_experiment_query(),
+            team=self.team,
+            timings=self.timings,
+            modifiers=create_default_modifiers_for_team(self.team),
+        )
+
+        if self.metric.metric_type == ExperimentMetricType.FUNNEL:
+            return [
+                ExperimentVariantFunnelsBaseStats(
+                    failure_count=result[1] - result[2],
+                    key=result[0],
+                    success_count=result[2],
+                )
+                for result in response.results
+            ]
+
+        return [
+            ExperimentVariantTrendsBaseStats(
+                absolute_exposure=result[1],
+                count=result[2],
+                exposure=result[1],
+                key=result[0],
+            )
+            for result in response.results
+        ]
+
+    def calculate(self) -> ExperimentQueryResponse:
+        variants = self._evaluate_experiment_query()
+
+        control_variant = next((variant for variant in variants if variant.key == CONTROL_VARIANT_KEY), None)
+        test_variants = [variant for variant in variants if variant.key != CONTROL_VARIANT_KEY]
+
+        if not control_variant:
+            raise ValueError("Control variant not found in experiment results")
+
+        # Statistical analysis
+        if self.stats_version == 2:
```

Comment:
Can't we only support `stats_version == 2` here?

Reply:
Yes, shouldn't be possible with stats_version == 1 here. Noted it down to clean this up.